views/js/runner/plugins/security/watermark.js

```diff
@@ -0,0 +1,441 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2025 (original work) Open Assessment Technologies SA ;
+ */
+
+define([
+    'jquery',
+    'lodash',
+    'taoTests/runner/plugin',
+    'context',
+    'taoQtiTest/runner/helpers/map',
+    'core/digest',
+    'ui/dialog',
+    'tpl!taoTestRunnerPlugins/runner/plugins/security/templates/watermarkCirclePath',
+    'css!taoTestRunnerPlugins/runner/plugins/security/css/watermark'
+], function ($, _, pluginFactory, context, mapHelper, digest, dialog, watermarkCirclePathTpl) {
+    'use strict';
+
+    const defaultConfig = {
+        /**
+         * "SHA-1" or "SHA-256" - hashing algorithm
+         * Here we don't use this hash for security checks, so simplest algorithm is fine?
+         *  @type {String}
+         */
+        hashAlgorithm: 'SHA-1',
+        /**
+         * 'background' | 'foreground' | 'circle'
+         * @type {String}
+         */
+        type: 'background',
```

Comment:
Isn't it mixing 2 concepts? I would expect `type` to be `'circle' | 'horizontal' | 'diagonal'` and `layer` to be `'background' | 'foreground'`. Then there would be 6 usable presets in case the customer changes their mind. (And I suppose it wouldn't be that much more complex to do?)